src/java/org/apache/cassandra/db/SinglePartitionReadCommand.java

```diff
@@ -428,6 +428,11 @@ public ClusteringIndexFilter clusteringIndexFilter(DecoratedKey key)
         return clusteringIndexFilter;
     }
 
+    @Override
+    public boolean isCheap() {
+        return limits().count() <= 1;
```

Comment:
The only notable change in this port is that cheapness is determined by the new limit feature (rather than pulling a count from the filter).